CMake: disable some warnings that have been introduced in LLVM 11

agarny opened the pull request:
Addresses issue #760.

agarny committed:
CMake: disable some warnings that have been introduced in LLVM 11.

agarny committed:
CMake: some minor cleaning up.

agarny commented:
@hsorby, @kerimoyle and @nickerso: a (very) quick one to review.

agarny commented:
Thanks @hsorby. One more, @kerimoyle and @nickerso. :)

kerimoyle commented:
Wasn't the deal that you were going to add documentation to this PR before it would go through?

agarny commented:
We agreed that documentation should be added (see issue #764), but as far as I know there was no agreement on whether this should be done before this PR or after. However, issue #765 probably ought to be addressed before this PR.

kerimoyle commented:
Haha, ok.  I reckon unless it's added now (when someone - that's you! - still cares about it) it won't ever happen! But yes, agreed that we need to sort out #765 first.

agarny committed:
Make ClangFormat happy.

This is the result of moving to LLVM/Clang 11.

agarny commented:
Ok, I have upgraded our macOS CI machine to use LLVM/Clang 11 (hence commit a9a9d34), so hopefully we should now be all good to review and merge this in.

nickerso commented:
The documentation issue is resolved and we should resend this one to get ticks across the board?

agarny commented:
> The documentation issue is resolved and we should resend this one to get ticks across the board?

Yes, I did that 2 days ago, but `brew upgrade` also updated Python and messed things up for us when it comes to buildbot's documentation. So, I fixed that (which was a bit crazy since it requires Python 3.7.7, but also whatever version of Python is available, 3.9 in our case) and, from what I can tell, everything is fine (at least when running the documentation from the command line on our macOS CI machine). However, Buildbot somehow stalled.

Anyway, I had planned to retrigger things this morning. So, hopefully, Buildbot is now going to behave itself.

agarny commented:
Ok, FWIW, to redeliver a payload doesn't get Buildbot to build/test the corresponding commit. So, am going to push a dummy commit to see whether it makes a difference. Ping @hsorby.

agarny committed:
Revert "Make ClangFormat happy."

This reverts commit a9a9d340fb6c3c03e12d03074b2e5cd1043b682a.

agarny committed:
CMake: some minor cleaning up.

agarny commented:
Ok, just to confirm that pushing a dummy commit did get Buildbot to trigger a job. Hopefully back on track!

agarny committed:
Tests: make MSVC happy.

agarny commented:
Ok, we are back on track. IOW, this PR is now ready for (re-)reviewing. Ping @hsorby, @kerimoyle and @nickerso.

kerimoyle commented on src/analyser.cpp:
```diff
@@ -282,7 +282,7 @@ bool AnalyserInternalEquation::check(size_t &equationOrder, size_t &stateIndex,
         if (variable->mType == AnalyserInternalVariable::Type::UNKNOWN) {
```
I'm not sure what changes in this file have to do with this PR?  Also the test_utils.cpp ones?

agarny replied:
For this file, it has to do with ClangFormat. In CF 11, they "properly" indent the ternary operator.

Regarding `tests/test_utils.cpp`, it's to prevent a warning on Windows.
